api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/EncounterTranslatorImpl.java

```diff
@@ -137,4 +148,12 @@ public org.openmrs.Encounter toOpenmrsType(@Nonnull org.openmrs.Encounter existi
 		
 		return existingEncounter;
 	}
+
+	private EncounterRole getDefaultEncounterRole() {
+		EncounterRole role = encounterService.getEncounterRoleByName("Unknown");
+		if (role == null) {
+			throw new IllegalStateException("Missing encounter role named 'Unknown'");
+		}
+		return role;
+	}
```

Comment:
I don't love hard-coding things like this. At the very least, this needs to be `protected` rather than `private` so it's overridable. Also, I would prefer this to be at least a GP or something the implementation can control in some form or fashion. Finally, ideally, setting the encounter role would be done in the participantTranslator and not here.

Reply:
Thanks for the feedback Ian, I've now moved this to participant translator.
Would it make sense trying to fetch role from GP, something like this:

```
	protected EncounterRole getDefaultEncounterRole() {
		String defaultEncounterRoleName = globalPropertyService.getGlobalProperty("fhir2.defaultEncounterRoleName");

		String roleName = (defaultEncounterRole != null && !defaultEncounterRole.isEmpty())
				? defaultEncounterRoleName
				: "Unknown";

		EncounterRole role = encounterService.getEncounterRoleByName(roleName);

		if (role == null) {
			throw new IllegalStateException("Missing encounter role named 'Unknown'");
		}

		return role;
	}
```
Without setting the encounter role an ex would occur when saving the encounter:
SQLIntegrityConstraintViolationException: Column 'encounter_role_id' cannot be null

Reply:
I understand that we need to fill in the field, it's just the module should be as agnostic about the system's metadata as possible. That's exactly the sort of solution I was imagining.

Reply:
Modified PR to fetch role from GP if present [here](https://github.com/openmrs/openmrs-module-fhir2/pull/542/files#diff-63c16fdd2f4cbcc348687f85367b9cb611f008db42e516063744c7617c69c334R70).
@ibacher please have a look and let me know if you have any change suggestion. Thanks!